packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -578,7 +578,7 @@ export async function destroyStack(options: DestroyStackOptions) {
 
   const currentStack = await CloudFormationStack.lookup(cfn, deployName);
   if (!currentStack.exists) {
-    return;
+    throw new StackNotFoundError(`Failed to destroy ${deployName}. The stack is not deployed.`);
```

Comment:
This is a breaking change, because we are changing the exit code for this command. We will not accept breaking changes to any CLI commands. Instead please a new flag, something `--fail-on-missing`, to `destroy`, and in this `if` ensure we check both `currentStack.exists` and `options.failOnMissing`, or a different name if you think a different one would be better.

Reply:
Would this be considered a breaking change even though the [exception is being caught](https://github.com/aws/aws-cdk/pull/30374/files/bc1db8f5b91f06d876110f3afcc83b3f806bb5da#diff-8cff125f4b53b025add8740ab74f39ee6c20221c0d3ac5ab7db184a6bb077398R623-R625)? The exit code of destroy command is not changed. The command doesn't fail if the stack doesn't exist.

The issue states `destroy` returns success even if the stack doesn't exist, if we add the flag this problem will persist with the `destroy` command.